src/renderer/services/dex.js

```diff
@@ -1276,110 +1281,110 @@ export default {
-                return false;
-              }
-              if (currentUnspent.reservedFor && currentUnspent.reservedFor.length >= 40) {
-                // reserved for someone else
-                return false;
-              }
-              const utxoKey = `${currentUnspent.txid}-${currentUnspent.index}-${currentNetwork.net}`;
-              if (_.has(assetUTXOsToIgnore, utxoKey)
-                && _.get(assetUTXOsToIgnore, utxoKey) >= currentNetwork.bestBlock.index) {
-                // we've tried to use this UTXO before and failed, skip it
-                // console.log(`We've tried to use this UTXO before and failed, skip it. unspent: ${JSON.stringify(currentUnspent)} `);
-                return false;
-              }
+        const dexAddress = wallet.getAddressFromScriptHash(currentNetwork.dex_hash);
 
-              if (quantitySumOfPickedInputs.isGreaterThanOrEqualTo(quantity)) {
-                let isDonePicking = true;
-                let i = 0;
-                pickedUnspents.some((pickedUnspent) => {
-                  if (quantitySumOfPickedInputs.minus(pickedUnspent.value).plus(currentUnspent.value)
-                    .isGreaterThanOrEqualTo(quantity)) {
-                    // remove pickedInput and use the current one.
-                    pickedInputs.splice(i, 1);
-                    pickedUnspents.splice(i, 1);
-                    quantitySumOfPickedInputs = quantitySumOfPickedInputs.minus(pickedUnspent.value);
-                    // console.log(`-$ removed input to use for withdraw total: ${quantitySumOfPickedInputs} unspent: ${JSON.stringify(pickedUnspent)}`);
-                    isDonePicking = false;
-                    return true;
-                  }
-                  i += 1;
-                  return false;
-                });
-                if (isDonePicking) {
-                  return true;
-                }
+        try {
+          config.balance = await store.dispatch('fetchSystemAssetBalances', { forAddress: dexAddress });
+        } catch (e) {
+          reject(`Failed to fetch address balance. ${e}`);
+          return;
+        }
+
+        const unspents = assetId === assets.GAS ? config.balance.assets.GAS.unspent : config.balance.assets.NEO.unspent;
+        await this.decorateWithUnspentsReservedState(assetId, unspents);
+
+        const pickedInputs = [];
+        const pickedUnspents = [];
+        let quantitySumOfPickedInputs = new BigNumber(0);
+        _.orderBy(unspents, [unspent => parseFloat(unspent.value.toString())], ['desc']).some((currentUnspent) => {
+          if (currentUnspent.reservedFor === currentWalletScriptHash) {
+            this.completeSystemAssetWithdrawals();
+            reject('Already have a UTXO reserved for your address. Completing open withdraw.');
+            return false;
+          }
+          if (currentUnspent.reservedFor && currentUnspent.reservedFor.length >= 40) {
```

Comment:
Can you throw `40` in a `const` with a descriptive name so we know what it is?

Reply:
Yes, in the future; it is 20 bytes in ascii hex (40 bytes). This happens to be the length of a user or contract's script hash. I'll make a constant for it as a follow up.